### ISSUE TITLE
Origin no longer closes upon exit.

### ISSUE THREAD

**Issue by Whalee110** [FriendsOfGalaxy/galaxy-integration-origin#12]
Origin no longer closes automatically once game is closed.

Launched origin game and closed around 10:20 AM to 10:26 AM within the log.

[plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log](https://github.com/FriendsOfGalaxy/galaxy-integration-origin/files/3947611/plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log)

**Comment by Whalee110** [comment]
Further testing I have found the client closes when playing Star Wars Jedi Fallen Order. When playing Dragon Age Inquisition the client does not close.

Star Wars was closed around 11:25:50 AM - A call for 'Client_Shutdown' can be seen.
Dragon Age Inquisition was closed around 11:28:30 AM - No references within the log point to the games closure, or a follow up call for 'Client_shutdown'

I have attached a new log with this information.

[plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log](https://github.com/FriendsOfGalaxy/galaxy-integration-origin/files/3947805/plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log)

**Comment by FriendsOfGalaxy** [comment]
Hey, looks like Galaxy asks for shutdown_platform_client after status changed notification (from "played" to "installed"). 

```
2019-12-11 11:16:50,584 - root - DEBUG - Local games cache is fresh enough
2019-12-11 11:16:51,595 - root - INFO - Sending notification: method=local_game_status_changed, params={'local_game': LocalGame(game_id='Origin.OFR.50.0003815', local_game_state=<LocalGameState.Installed: 1>)}
2019-12-11 11:16:51,595 - root - DEBUG - Sending 145 byte of data
2019-12-11 11:16:51,595 - root - DEBUG - Task manager notification client: creating task 94 (send)
2019-12-11 11:16:51,595 - root - DEBUG - Task manager notification client: finished task 94 (send)
2019-12-11 11:16:51,615 - root - DEBUG - Received 65 bytes of data
2019-12-11 11:16:51,615 - root - INFO - Handling notification: method=shutdown_platform_client, params={}
```

Can you confirm that after Dragon Age Inquisition was closed, Galaxy still thinks it is running? The Play button should be gray. If so, can you check it the game process it really closed?

**Comment by Whalee110** [comment]
Thanks for the reply. I have checked again -  once closed the game is able to be relaunched from GOG 2.0 (So GOG 2.0 knows it has closed) - I tried closing all additional services on Origin to see if it was a component there causing it but no luck.

So far 2 games it doesn't work for, and only Star wars does it pick up.

Is this a bug what will need to be fixed? Or should I test reinstalling Origin / completely wiping the plugin cache and reinstalling?

**Comment by Whalee110** [comment]
An update!

Recently I reinstalled Origin, but didn't reinstall games as the files were already there. In that time Battlefield V and Star Wars: Jedi Fallen Order had updated / installed. These two games were fine - I did a quick repair on Dragon Age Inquisition and The Saboteur. Before the repair they both didn't close Origin, after the repair they did. I was looking in the registry and couldn't find any new entries, but they may have updated some origin config files.

Origin now closes after exiting all games I currently have installed, I have attached a log if it helps, but it may be a note to repair games if experience similar in the future :)

Thanks for your help!
[plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log](https://github.com/FriendsOfGalaxy/galaxy-integration-origin/files/3953340/plugin-origin-7f53219b-4e2b-4591-9f4f-dfc5f4ba9eb0.log)
